drivers/tapdisk-nbdserver.c

```diff
@@ -1088,30 +1095,62 @@ tapdisk_nbdserver_newclient_fd(td_nbdserver_t *server, int new_fd)
 	}
 }	
 
-void
-tapdisk_nbdserver_clientcb(event_id_t id, char mode, void *data)
+static td_vbd_request_t *create_request_vreq(
+	td_nbdserver_client_t *client, struct nbd_request request, uint32_t len)
 {
-	td_nbdserver_client_t *client = data;
-	td_nbdserver_t *server = client->server;
 	int rc;
-	int len;
-	int fd = client->client_fd;
+	td_nbdserver_t *server = client->server;
 	td_vbd_request_t *vreq;
-	struct nbd_request request;
 	td_nbdserver_req_t *req;
 
 	req = tapdisk_nbdserver_alloc_request(client);
 	if (!req) {
 		ERR("Couldn't allocate request in clientcb - killing client");
-		tapdisk_nbdserver_free_client(client);
-		return;
+		goto failreq;
 	}
 
 	vreq = &req->vreq;
 
 	memset(req, 0, sizeof(td_nbdserver_req_t));
-	/* Read the request the client has sent */
 
+	bzero(req->id, sizeof(req->id));
```

Comment:
Why do you need to zero req->id when the whole of req was zeroed in the previous line?

Reply:
we don't but that was a pre-existing issue from the looks of it.